crates/cli-support/src/js.rs

```diff
@@ -316,42 +317,42 @@ impl<'a> Context<'a> {
             ts_dst.push_str("
                 public ptr: number;
             ");
-            if self.config.debug {
-                self.expose_check_token();
-                dst.push_str(&format!("
-                    constructor(ptr, sym) {{
-                        _checkToken(sym);
-                        this.ptr = ptr;
+
+            self.expose_constructor_token();
+
+            dst.push_str(&format!("
+                constructor(...args) {{
+                    if (args.length === 1 && args[0] instanceof ConstructorToken) {{
```

Comment:
Could this binding code be generated only if the constructor is present? If it's not present I think the previous construction should work, right?

Reply:
Yes, I've changed it so that Token version is only used when either debug or a constructor are set. Direct construction is possible through a tiny `__construct` shim. This also allows functions to directly create an instance even if they don't know whether the struct has a constructor or not. (I can change the name if you've got a better one)